test/org/zaproxy/zap/extension/websocket/client/ServerConnectionEstablisherUnitTest.java

```diff
@@ -0,0 +1,109 @@
+	}
+	
+	@Test
+	public void shouldReceiveUpgradeStatusCode() throws Exception {
+		ServerConnectionEstablisher establisher = new ServerConnectionEstablisher();
+		HttpMessage handshakeRequest = new HttpMessage(HttpHandshakeBuilder.getHttpHandshakeRequestHeader(new URI("http://" + HOST_NAME + ":" + WEBSOCKET_PORT,true)));
+		NanoWebSocketTestServer webSocketServer = super.getWebSocketTestServer();
+		System.out.println(handshakeRequest.getRequestHeader().toString());
+		establisher.send(new HandshakeConfig(handshakeRequest,false,false));
+		assertEquals(101,handshakeRequest.getResponseHeader().getStatusCode());
+		assertEquals(webSocketServer.getLastConnection()
+						.getHandshakeResponse().getHeader(HttpHandshakeBuilder.SEC_WEB_SOCKET_ACCEPT)
+				, WebSocketUtils.encodeWebSocketKey(handshakeRequest.getRequestHeader().getHeader(HttpHandshakeBuilder.SEC_WEB_SOCKET_KEY)));
+		
+	}
+	
+	@Test
+	public void shouldReturnWSProxy() throws Exception {
+		ServerConnectionEstablisher establisher = new ServerConnectionEstablisher();
+		HttpMessage handshakeRequest = new HttpMessage(HttpHandshakeBuilder.getHttpHandshakeRequestHeader(new URI("http://" + HOST_NAME + ":" + WEBSOCKET_PORT,true)));
+		
+		WebSocketProxy webSocketProxy = establisher.send(new HandshakeConfig(handshakeRequest,false,false));
+		NanoWebSocketConnection webSocketConnection = super.getWebSocketTestServer().getLastConnection();
+		assertEquals(101,handshakeRequest.getResponseHeader().getStatusCode());
+		assertNotNull(webSocketProxy);
+		assertTrue(webSocketProxy.isConnected());
+		assertEquals(webSocketConnection.getHandshakeResponse().getHeader(HttpHandshakeBuilder.SEC_WEB_SOCKET_ACCEPT)
+				,WebSocketUtils.encodeWebSocketKey(handshakeRequest.getRequestHeader().getHeader(HttpHandshakeBuilder.SEC_WEB_SOCKET_KEY)));
+	}
+	
+	@Test
+	public void shouldReceiveMessagesFromServer() throws Exception {
+		ServerConnectionEstablisher establisher = new ServerConnectionEstablisher ();
+		HttpMessage handshakeRequest = new HttpMessage(HttpHandshakeBuilder.getHttpHandshakeRequestHeader(new URI("http://" + HOST_NAME + ":" + WEBSOCKET_PORT,true)));
+		
+		WebSocketProxy webSocketProxy = establisher.send(new HandshakeConfig(handshakeRequest,false,false));
+		NanoWebSocketConnection webSocketConnection = super.getWebSocketTestServer().getLastConnection();
+		
+		Stack<NanoWSD.WebSocketFrame> webSocketFrameStack = new Stack<>();
+		webSocketFrameStack.push(new NanoWSD.WebSocketFrame(NanoWSD.WebSocketFrame.OpCode.Text,true,"Hello World"));
+		webSocketFrameStack.push(new NanoWSD.WebSocketFrame(NanoWSD.WebSocketFrame.OpCode.Text,true,"Hello World-2"));
+		webSocketFrameStack.push(new NanoWSD.WebSocketFrame(NanoWSD.WebSocketFrame.OpCode.Text,true,"Hello World-3"));
+		webSocketConnection.setOutgoingMessageSchedule(webSocketFrameStack,50);
+		
+		Thread.sleep(200);
+		assertEquals(4,webSocketProxy.getIncrementedMessageCount());
+		
+	}
+	
+	@After
+	public void stopWebSocketServer(){
```

Comment:
- Better if this is added at the beginning of the class (after `@Before`), easier to follow the set up and tear down.
 - It should have an `@Override` annotation.

(Same comments for following class.)